kubernetes/resource_kubernetes_default_service_account.go

```diff
@@ -55,16 +55,22 @@ func resourceKubernetesDefaultServiceAccountCreate(ctx context.Context, d *schem
 		log.Printf("[INFO] Default service account exists: %s", metadata.Namespace)
 		return nil
 	})
-
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	secret, err := getServiceAccountDefaultSecret(ctx, "default", svcAcc, d.Timeout(schema.TimeoutCreate), conn)
+	var defaultSecret *api.Secret
+	b, err := serverVersionLessThan(conn, "1.24.0")
 	if err != nil {
 		return diag.FromErr(err)
 	}
-	d.Set("default_secret_name", secret.Name)
+	if b {
```

Comment:
Silently skipping setting any value on `default_secret_name` might be confusing to users with existing configurations.
I would suggest returning a Warning level Diagnostic when we do this, to alert users in the Terraform output about the new behaviour.

Reply:
I have updated the logic and moved the condition statement that verifies the version a bit lower. Now we should get a warning when creating or updating any of `r/kubernetes_default_service_account`, `d/kubernetes_service_account`, and `r/kubernetes_service_account`, however, when we do the import for `r/kubernetes_service_account` -- we don't get a warning message, because import expects only an error as return, and this is not an error.